pallets/avn/src/vote.rs

```diff
@@ -32,20 +34,21 @@ pub const VOTING_SESSION_DATA_IS_NOT_FOUND: u8 = 5;
 pub const APPROVE_VOTE: bool = true;
 pub const REJECT_VOTE: bool = false;
 
-#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, TypeInfo)]
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, TypeInfo, MaxEncodedLen)]
 pub struct VotingSessionData<AccountId, BlockNumber> {
+    // TODO convert to BoundedVec
     /// The unique identifier for this voting session
-    pub voting_session_id: Vec<u8>,
+    pub voting_session_id: WeakBoundedVec<u8, VotingSessionIdBound>,
```

Comment:
Why not convert to boundedVec now?

Reply:
Won't this need a data migration? Because we are changing the inner type of a struct which will change how the data is encoded.

Reply:
> Why not convert to boundedVec now?

Next PR handles that.

> Won't this need a data migration? Because we are changing the inner type of a struct which will change how the data is encoded.

Good question. It is actually safe. The bounded vector implementation is the following:
```
/// A bounded vector.
///
/// It has implementations for efficient append and length decoding, as with a normal `Vec<_>`, once
/// put into storage as a raw value, map or double-map.
///
/// As the name suggests, the length of the queue is always bounded. All internal operations ensure
/// this bound is respected.
#[cfg_attr(feature = "std", derive(Serialize), serde(transparent))]
#[derive(Encode, scale_info::TypeInfo)]
#[scale_info(skip_type_params(S))]
pub struct BoundedVec<T, S>(
	pub(super) Vec<T>,
	#[cfg_attr(feature = "std", serde(skip_serializing))] PhantomData<S>,
);
```
BoundedVec is a tuple struct of Vector and PhantomData.
PhantomData is a marker type so it doesn't get included during the scale encoding of the data.

The `skip_type_params` attribute also skips the field from generating type information for the field as well.
https://docs.rs/scale-info/latest/scale_info/#scale_infoskip_type_params

The `serde(skip_serializing)` attribute excludes the field when performing a serde serialisation.

From my understanding the encoding excludes the const type that applies the bound, and the code implementation is responsible to use the correct bound otherwise the decoding will fail.

In sort the encoding is the same as having a vector, and there isn't a concern when you change it nor a data migration is needed.
We can put this to the test:
```
    #[test]
    fn encoding_tests() {
        let vector: Vec<u8> = vec![1,2,3];
        let bounded = BoundedVec::<u8,ConstU32<3>>::truncate_from(vector.clone());
        assert_eq!(vector.encode(), bounded.encode());
    }
```
I have also successfully upgrade the chain with summary data. Apart from a UI refresh the chain had no issue with the upgrade